server/connectors/bitbucket_connector/api_client.py

```
@@ -11,6 +11,14 @@
 logger = logging.getLogger(__name__)
 
 BITBUCKET_API_BASE = "https://api.bitbucket.org/2.0"
+_BITBUCKET_ALLOWED_HOSTS = frozenset({"api.bitbucket.org", "bitbucket.org"})
+
+
+def _validate_bitbucket_url(url: str) -> None:
+    """Raise ValueError if url does not point to a known Bitbucket host."""
+    host = urlsplit(url).hostname
+    if host not in _BITBUCKET_ALLOWED_HOSTS:
+        raise ValueError(f"URL host '{host}' is not a known Bitbucket domain")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)